pkg/runtime/wfengine/workflow.go

```diff
@@ -440,6 +449,8 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 				WithContentType(invokev1.OctetStreamContentType)
 			defer req.Close()
 
+			wf.activityResultAwaited = true
```

Comment:
will this state be accessed by multiple goroutines? Does this need to be an atomic boolean?